src/Components/Server/src/Circuits/RemoteRenderer.cs

```diff
@@ -79,6 +83,44 @@ public Task AddComponentAsync(Type componentType, string domElementSelector)
             return RenderRootComponentAsync(componentId);
         }
 
+        protected override void ProcessRenderQueue()
+        {
+            if (_unacknowledgedRenderBatches.Count >= _options.MaxBufferedUnacknowledgedRenderBatches)
+            {
+                // If we got here it means we are at max capacity, so we don't want to actually process the queue,
+                // as we have a client that is not acknowledging render batches fast enough (something we consider needs
+                // to be fast).
+                // The result is somethign as follows:
+                // Lets imagine an extreme case where the server produces a new batch every milisecond.
+                // Lets say the client is able to ACK a batch every 100 miliseconds.
+                // When the app starts the client might see the sequence 0.000->0.{MAXUnacknowledgeRenderBatches} and then
+                // after 100 miliseconds it sees it jump to 0.1xx, then to 0.2xx where xx is something between {0..99} the
+                // reason for this is that the server slows down rendering new batches to as fast as the client can consume
+                // them.
+                // Similarly, if a client were to send events at a faster pace than the server can consume them, the server
+                // would still proces the events, but would not produce new renders until it gets an ack that frees up space
+                // for a new render.
+                // We should never see UnacknowledgedRenderBatches.Count > _options.MaxBufferedUnacknowledgedRenderBatches
+
+                // But if we do, it's safer to simply disable the rendering in that case too instead of allowing batches to
+                // keep piling up.
+                if (!_queueIsFullNotified)
+                {
+                    // We skip logging after the first time we run into this situation to avoid filling the logs with
+                    // entries on high frequency render scenarios.
+                    // We try our best to log the queue is full at most once, although we don't guarantee it due to
+                    // concurrency.
+                    // When we complete a render we set this flag to false, so the next time the queue gets filled it
+                    // also shows up.
+                    _queueIsFullNotified = true;
+                    Log.FullUnacknowledgedRenderBatchesQueue(_logger);
+                }
```

Comment:
I don't really think we have a concern with this. Like we tell you not to have debug logging on in production unless you're investigating a specific problem.

Reply:
If not, it makes a lot of these things really hard to troubleshoot

Reply:
Got rid of the if statement, we can add it later if people complain I guess